File: C-Codes/triangle.c
#include<stdio.h>
#include<math.h>
#define e  10^-4

int main()
{
 float a,b,c;
 
 printf("Enter the sides of triangle=  ");
 scanf("%f %f %f", &a, &b, &c);
 printf("Sides are %f %f %f\n ", a, b, c);

 if (((a + b) <= c) || ((b + c) <= a) || ((a + c) <= b))
   printf("Triangle is not possible\n");
 else 
   {
    if (a == b == c )
   printf("Triangle is equilateral\n");
    else 
     if (a == b || b == c || c == a)
      printf("Triangle is isoscales only\n");
    else
     if((fabs(a*a - b*b + c*c) < e || fabs(b*b - a*a + c*c) < e || fabs(c*c == a*a + b*b) < e) && (a == b || b == c || c == a))
       printf("Triangle is isoscales and right angled also\n");
     else
      if((fabs(a*a - b*b + c*c) < e || fabs(b*b - a*a + c*c) < e|| fabs(c*c == a*a + b*b) < e ) && (a != b != c))
       printf("Triangle is scalene and  right angled also\n");
      else
       printf("Triangle is scalene only\n");
   }
return 0;
}

File: Python-Codes/BasicCodes/triangle.py
from math import *
def tri(a,b,c):   
	eps = 1.0e-12
	#a = input("enter one side of triangle : ")
	#b = input("enter one side of triangle : ")
	#c = input("enter one side of triangle : ")
	if (((a + b - c) < eps) or ((b + c - a) < eps) or ((a + c - b) < eps)):
		print("Triangle is not possible")
	else : 	
		if (((fabs(a-b) < eps) and fabs(a-c) != 0) or ((fabs(b-c) < eps) and fabs(a-c) != 0) or ((fabs(a-c) < eps) and fabs(b-c) != 0)):
			print("Triangle is isosceles")
			if (fabs((a**2-(b**2+c**2))) < eps or fabs((b**2-(a**2+c**2))) < eps or fabs((c**2-(b**2+a**2)))<eps):
				print("Right angle triangle also")
		elif ((fabs(a-b) < eps) and (fabs(b-c) < eps) and (fabs(a-c) < eps)):
			print("Triangle is equilateral")
		else :
			print("Triangle is scalene")
			if (fabs(((a**2-(b**2+c**2))) < eps) or (fabs((b**2-(a**2+c**2))) < eps) or (fabs((c**2-(b**2+a**2)))<eps)):
				print("Right angle triangle also")
			
			
		A = degrees(acos((c**2 + b**2 - a**2)/float(2*b*c)))
		B = degrees(acos((a**2 + c**2 - b**2)/float(2*a*c)))
		C = degrees(acos((a**2 + b**2 - c**2)/float(2*a*b)))
		print A,B,C
